q2_anomaly_detection/models.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+
+class Cloud():
+    def __init__(self, n_neighbors="auto", metric="precomputed", percentile_cutoff=0.95):
+        """
+        n_neighbors: Interger/String default='auto'
+            'auto' will set k = 10% of reference set size
+        """
+        self.k = n_neighbors
+        self.metric=metric
+        self.percentile_cutoff = percentile_cutoff
+        
+    def fit(self, X):
+        """
+        x: 2-d np.array
+            Distances between each sample in the reference set. The distance between a
+            sample and itself should be zero. Note: X should be square
+            where row i represents sample i and column j represents sample j. So,
+            X[i,j] should be the distance between sample i and j and X[i, i] = 0.
+        """
+        if self.k == "auto":
+            k = int(X.shape[0] * 0.1) 
+            self.k = k if k > 0 else 1
+        
+        if self.metric == "precomputed":
+            self.di = np.apply_along_axis(
+                # want to find the k nearest neighbors
+                # however, the distance  between a sample and itself will always
+                # be included (i.e. has a distance of 0). Thus we want to find
+                # the k+1 nearest neighbors.
+                lambda x: x[np.argpartition(x, self.k+1)[:self.k+1]].sum() / self.k,
+                axis=1,
+                arr=X
+            )
+
+        self.d_bar = self.di.mean()
+        self.ri = self.di / self.d_bar
+    
+    def score_samples(self, X):
```

Comment:
We want this to take the convention that lower score means more likely to be an anomaly, I think this should return `-rj`